meta_error.go

```diff
@@ -0,0 +1,66 @@
+package microerror
+
+import (
+	"encoding/json"
+)
+
+// MetaError is a predefined error structure which purpose is to act as
+// container for meta information associated to a specific error. The specific
+// error type matching can be used as usual. The usual error masking and cause
+// gathering can be used as usual. Using MetaError might look as follows. In the
+// beginning is a usual error defined, along with its matcher. This error is the
+// root cause once emitted during runtime.
+//
+//     var notEnoughWorkersError = microerror.MetaError{
+//         Desc: "The amount of requested guest cluster workers exceeds the available number of host cluster nodes.",
+//         Docs: "https://github.com/giantswarm/ops-recipes/blob/master/349-not-enough-workers.md",
+//         Kind: "notEnoughWorkersError",
+//     }
+//
+//     func IsNotEnoughWorkers(err error) bool {
+//         return microerror.Cause(err) == notEnoughWorkersError
+//     }
+//
+type MetaError struct {
```

Comment:
I think `microerror.Error` (vs. `microerror.MetaError) is enough as we don't have plans to introduce other types of errors.

Reply:
Done.